test/solvers/test_trust_regions_hessian_update.jl

```diff
@@ -0,0 +1,181 @@
+using Manopt, Manifolds, ManifoldsBase, LinearAlgebra, Test
+import Random: seed!
+
+@testset "Riemannian Trust-Region with Hessian update" begin
+    n = 100
+
+    A = randn(n, n)
+    A = (A + A') / 2
+    cost(::Sphere, p::Array{Float64,1}) = p' * A * p
+    grad(::Sphere, p) = 2 * (A * p - p * p' * A * p)
+    function hess(::Sphere, p::Array{Float64,1}, X::Array{Float64,1})
+        return 2 * (A * X - p * p' * A * X - X * p' * A * p - p * p' * X * p' * A * p)
+    end
+    grad!(::Sphere, X, p) = (X .= 2 * (A * p - p * p' * A * p))
+    function hess!(::Sphere, X, p, Y)
+        X .= 2 * (A * Y - p * p' * A * Y - Y * p' * A * p - p * p' * Y * p' * A * p)
+        return X
+    end
+    M = Sphere(n - 1)
+    x = random_point(M)
```

Comment:
Please do not use random points in tests. If we have a bug that only sometimes happens, and this makes this test sometimes fail and not fail – we are in the same setting as for WolfePowell and Tests are super hard to debug if there is randomness in there.

Maybe just take the first unit vector – or even a smaller manifold, we do not have to beat benchmarks in tests, `n=7` is fine as well.